From be9e904997b399383aaac6a479c340d4a6fab0d7 Mon Sep 17 00:00:00 2001
From: jjacky <i.am.jack.mail@gmail.com>
Date: Wed, 24 Apr 2013 11:27:59 +0200
Subject: [PATCH] treestore: Fix segfault when reordering non-existing children

Calling gtk_tree_store_reorder() on a node without children would cause a
segfault; Fixes https://bugzilla.gnome.org/show_bug.cgi?id=698396

Signed-off-by: Olivier Brunel <jjk@jjacky.com>
---
 gtk/gtktreestore.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/gtk/gtktreestore.c b/gtk/gtktreestore.c
index bbf6e5d..6db1f56 100644
--- a/gtk/gtktreestore.c
+++ b/gtk/gtktreestore.c
@@ -2285,6 +2285,12 @@ gtk_tree_store_reorder (GtkTreeStore *tree_store,
   else
     level = G_NODE (parent->user_data)->children;
 
+  if (G_UNLIKELY (!level))
+    {
+      g_warning ("%s: Cannot reorder, parent has no children", G_STRLOC);
+      return;
+    }
+
   /* count nodes */
   node = level;
   while (node)
-- 
1.8.4

